docs/resources/samples/performance.md

```diff
@@ -0,0 +1,151 @@
+---
+title: 'Performance optimization'
+description: 'Learn how to optimize read and write performance in Office Scripts.'
+ms.date: 03/03/2021
+localization_priority: Normal
+---
+
+# Read and write performance optimization
+
+## Basic perf optimization
+
+See the [getting started performance section](getting-started.md#basic-performance-considerations) for the basics of Office Scripts.
+
+## Large data write scenario
+
+The `setValues()` Range API allows setting the values of a range. This API has data limitations depending on various factors such as data size, network settings, etc. In order to reliably update a large range of data, you'll need to think about doing data updates in smaller chunks. This script attempts to do this and writes rows of a range in chunks so that if a large range needs to be updated, it can be done in smaller parts. **Warning**: It has not been tested across various sizes so be aware of that if you want to use this in your script. As we have opportunity to test, we'll update with findings around how it performs for various data sizes.
```

Comment:
We should test this scenario before publicly releasing this sample.

Reply:
Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)